homeassistant/components/shelly/sensor.py

```diff
@@ -176,35 +171,14 @@
     "uptime": RestAttributeDescription(
         name="Uptime",
         device_class=sensor.DEVICE_CLASS_TIMESTAMP,
+        default_enabled=False,
         path="uptime",
     ),
 }
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up sensors for device."""
-
-    wrapper: ShellyDeviceWrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][
```

Comment:
Why are you removing the logic for battery sensors?

Reply:
This has moved to by a property of the sensor same as the inptu sensors:
```python
("device", "battery"): BlockAttributeDescription(
        name="Battery", unit=PERCENTAGE, device_class=sensor.DEVICE_CLASS_BATTERY	        name="Battery",
        unit=PERCENTAGE,
        device_class=sensor.DEVICE_CLASS_BATTERY,
        removal_condition=lambda settings, _: settings.get("external_power") == 1,
    ),
```